src/util.c

```diff
@@ -1380,3 +1381,25 @@ int snprintf_async_signal_safe(char *to, size_t n, const char *fmt, ...) {
     va_end(args);
     return result;
 }
+
+/*
+ * This function is similar to asprintf function, but it uses zmalloc for
+ * allocating the string buffer.
+ *
+ * IMPORTANT: don't forget to free the buffer when no longer needed.
```

Comment:
I'm not a huge fan of all-caps warnings to the reader. :)

If we just say "returns a freshly allocated string", it should be enough for the reader to know it needs to be free'd.

```suggestion
/* A printf-like function that returns a freshly allocated string.
 *
 * This function is similar to asprintf function, but it uses zmalloc for
 * allocating the string buffer. */
```